server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java

```diff
@@ -419,35 +419,42 @@ public GlobalSession readSession(String xid) {
      */
     @Override
     public List<GlobalSession> readSession(GlobalStatus[] statuses, boolean withBranchSessions) {
-        List<String> statusKeys = new ArrayList<>();
-        for (int i = 0; i < statuses.length; i++) {
-            statusKeys.add(buildGlobalStatus(statuses[i].getCode()));
-        }
-        int limit = resetLogQueryLimit(statusKeys);
-        try (Jedis jedis = JedisPooledFactory.getJedisInstance()) {
-            Pipeline pipelined = jedis.pipelined();
-            statusKeys.stream().forEach(statusKey -> pipelined.lrange(statusKey, 0, limit));
-            List<List<String>> list = (List<List<String>>)(List)pipelined.syncAndReturnAll();
-            pipelined.close();
-            List<GlobalSession> globalSessions = Collections.synchronizedList(new ArrayList<>());
-            if (CollectionUtils.isNotEmpty(list)) {
-                List<String> xids = list.stream().flatMap(ll -> ll.stream()).collect(Collectors.toList());
-                xids.parallelStream().forEach(xid -> {
-                    GlobalSession globalSession = this.readSession(xid, withBranchSessions);
-                    if (globalSession != null) {
-                        globalSessions.add(globalSession);
-                    }
-                });
-            }
+
+        List<GlobalSession> globalSessions = Collections.synchronizedList(new ArrayList<>());
+        List<String> statusKeys = convertStatusKeys(statuses);
+
+        Map<String, Long> targetMap = calculateStatuskeysHasData(statusKeys);
+        long limit = resetLogQueryLimit(targetMap);
+        if (targetMap.size() == 0 || logQueryLimit <= 0) {
             return globalSessions;
         }
+        // totalCount
+        final long totalCount = Math.min(logQueryLimit, countByClobalSesisons(statuses));
```

Comment:
why use `countByClobalSesisons` but not targetMap's values sum?

Reply:
totalCount -> queryCount

Reply:
get